src/clp_ffi_js/ir/StructuredIrStreamReader.cpp

```diff
@@ -94,16 +90,14 @@ auto StructuredIrStreamReader::deserialize_stream() -> size_t {
     constexpr size_t cDefaultNumReservedLogEvents{500'000};
     m_deserialized_log_events->reserve(cDefaultNumReservedLogEvents);
     auto& reader{m_stream_reader_data_context->get_reader()};
-    while (true) {
-        auto result{m_stream_reader_data_context->get_deserializer().deserialize_next_ir_unit(reader
-        )};
+    auto& deserializer = m_stream_reader_data_context->get_deserializer();
+
+    while (false == deserializer.is_stream_completed()) {
+        auto result{deserializer.deserialize_next_ir_unit(reader)};
         if (false == result.has_error()) {
             continue;
         }
         auto const error{result.error()};
-        if (std::errc::operation_not_permitted == error) {
-            break;
-        }
         if (std::errc::result_out_of_range == error) {
```

Comment:
@LinZhihao-723 could you help confirm if this is the correct / preferred call sequence to handle End of Stream?

Reply:
- We don't need to check any error code on the end of stream; the end of stream is handled as an IR unit; you can check the returned IR unit type to confirm (but that should also be a part of the job of the unit handler)
- This error code was added to check truncated IR, which is the incomplete stream error. It's been fixed in ffi core, so only `std::errc::result_out_of_range` is needed: https://github.com/y-scope/clp/pull/579

Btw, u can refer to this part of code in clp-ffi-py to see some similar usage: https://github.com/y-scope/clp-ffi-py/blob/main/src/clp_ffi_py/ir/native/PyDeserializer.cpp#L224